### PR TITLE
Replace local broadcast deprecated

### PR DESCRIPTION
**Pull request by chihung93**
I have replaced localbroadcastmanager with Flow

The reason: 
https://developer.android.com/jetpack/androidx/releases/localbroadcastmanager
This artifact and its classes are deprecated. Use LiveData or reactive streams instead.

### CODE REVIEW

**Review by hannesa2 — CHANGES_REQUESTED** [review]
Awesome !
Please format code and test should be fine. But sometimes a re-trigger CI made it work.
I already did the re.trigger, but without success

File: serviceLibrary/src/main/java/info/mqtt/android/service/MqttAndroidClient.kt [hunk 1]
```diff
@@ -64,8 +72,7 @@ class MqttAndroidClient @JvmOverloads constructor(
     private var traceCallback: MqttTraceHandler? = null
     private var traceEnabled = false
 
-    @Volatile
-    private var receiverRegistered = false
+    private val receiverRegistered = MutableStateFlow(true)
```
[the remark this answers is not in the document]
Comment (chihung93) [reply serviceLibrary/src/main/java/info/mqtt/android/service/MqttAndroidClient.kt]: I have fixed it . I replaced it with MutableStateFlow<Boolean?>(null)

File: serviceLibrary/src/main/java/info/mqtt/android/service/MqttAndroidClient.kt [hunk 2]
```diff
@@ -218,29 +224,35 @@ class MqttAndroidClient @JvmOverloads constructor(
...
             pool.execute {
                 doConnect()
-
-                //Register receiver to show shoulder tap.
-                if (!receiverRegistered) {
-                    registerReceiver(this@MqttAndroidClient)
+            }
+        }
+        CoroutineScope(Dispatchers.IO).launch {
+            mqttService?.localBroadcastFlow?.filter { receiverRegistered.value && it.extras !=null && !it.extras!!.isEmpty }?.collectLatest { intent->
```
[the remark this answers is not in the document]
Comment (chihung93) [reply serviceLibrary/src/main/java/info/mqtt/android/service/MqttAndroidClient.kt]: I have added receiverRegistered.value != true

**Comment by chihung93** [comment]
> Please rebase, then it checks Kotlin code format

I have fixed your comment. Thanks

**Comment by hannesa2** [comment]
The Espresso tests show us, that there is an error introduced by this pull request

**Pull request closed without merging**